syntax = "proto3";

package Acoustid.Server.PB;

message SearchResult {
    uint32 doc_id = 1;
    float score = 2;
};

message SearchRequest {
    string index_name = 1;
    repeated uint32 terms = 2;
    int32 max_results = 3;
};

message SearchResponse {
    repeated SearchResult results = 1;
};

service Index {
    rpc Search(SearchRequest) returns (SearchResponse);
};
